src/plugins/discourse/references.js

```diff
@@ -23,3 +48,37 @@ export function parseLinks(cookedHtml: string): Hyperlink[] {
   parser.end();
   return links;
 }
+
+export function linksToReferences(
+  links: $ReadOnlyArray<Hyperlink>
+): DiscourseReference[] {
+  const server = "(https://[\\w.-]+)";
+  const topic = `${server}/t/[\\w-]+/(\\d+)`;
+  const post = `${topic}/(\\d+)`;
+  const params = "(?:\\?[\\w-=]+)?";
```

Comment:
Params may need the `%` and `+` included for url encoding and the alias for spaces.

Reply:
Or just `decodeURI` on the whole URL first?

Reply:
Sure, I added decodeURI. I don't know of any "real" examples (i.e. actual Discourse links) that need it, but can't hurt.